cmd/kubelet/app/server.go

```diff
@@ -288,6 +299,41 @@ is checked every 20 seconds (also configurable with a flag).`,
 	return cmd
 }
 
+// mergeKubeletConfigurations merges the provided drop-in configurations with the base kubelet configuration.
+func mergeKubeletConfigurations(kubeletConfig *kubeletconfiginternal.KubeletConfiguration, kubeletFlags *options.KubeletFlags) error {
+	const dropinFileExtension = ".conf"
+
+	// Load additional dropins, if provided
+	if !utilfeature.DefaultFeatureGate.Enabled(features.KubeletDropInConfig) {
+		return fmt.Errorf("flag %s specified but featuregate %s missing, cannot start kubelet", kubeletFlags.KubeletDropinDir, features.KubeletDropInConfig)
+	}
```

Comment:
nonblocking: usually the FG check is done at the highest possible layer, so I'd say in the calling site rather than inside this function.